module/zfs/dmu.c

```diff
@@ -1293,7 +1293,7 @@ dmu_write_impl(dmu_buf_t **dbp, int numbufs, uint64_t offset, uint64_t size,
 
 void
 dmu_write(objset_t *os, uint64_t object, uint64_t offset, uint64_t size,
-    const void *buf, dmu_tx_t *tx)
+    const void *buf, dmu_tx_t *tx, dmu_flags_t flags)
```

Comment:
My only concern is Lustre makes use of this interface so it'll break that build.  We could instead create a `dmu_write_flags` wrapper like we've done elsewhere.  That said, Lustre already makes use of the `_by_dnode()` variant so I don't think it'd be a big deal for them to switch to that.

Reply:
Most of other functions already take the flags argument, while `dmu_buf_will_dirty_flags()` is the only exception when we have a separate function for that.  I could make what you ask, but it would be an oddity.